hazelcast/src/main/java/com/hazelcast/map/impl/operation/BasePutOperation.java

```diff
@@ -37,8 +37,7 @@ public BasePutOperation() {
 
     @Override
     protected void afterRunInternal() {
-        Object value = isPostProcessing(recordStore)
-                ? recordStore.getRecord(dataKey).getValue() : dataValue;
+        Object value = getValueForPublishing();
         mapServiceContext.interceptAfterPut(mapContainer.getInterceptorRegistry(), dataValue);
         mapEventPublisher.publishEvent(getCallerAddress(), name, getEventType(),
                 dataKey, oldValue, value);
```

Comment:
Another way can be passing both de-serialized and serialized value to publish event method.
By doing this, we can eliminate extra serialization of value while sending event to remote machines.

Reply:
Thats right, How would you recommend we do that?  We can  add a new parameter to `publishEvent`.  Is it ok to change that api?

Reply:
Yes it is internal api, you can change it.